src/common.browser/AudioFileWriter.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as fs from "fs";
+import { IAudioDestination } from "../common/Exports";
+import { AudioOutputFormatImpl } from "../sdk/Audio/AudioOutputFormat";
+import { Contracts } from "../sdk/Contracts";
+import { AudioStreamFormat } from "../sdk/Exports";
+
+export class AudioFileWriter implements IAudioDestination {
+    private privAudioFormat: AudioOutputFormatImpl;
+    private privFd: number;
+    private privOffset: number = 0;
+    private privId: string;
+
+    public constructor(filename: fs.PathLike) {
+        this.privFd = fs.openSync(filename, "w");
+        this.privOffset = 0;
+    }
+
+    public set format(format: AudioStreamFormat) {
+        Contracts.throwIfNotUndefined(this.privAudioFormat, "format is already set");
+        this.privAudioFormat = format as AudioOutputFormatImpl;
+        if (this.privAudioFormat.hasHeader) {
+            this.privOffset = this.privAudioFormat.header.byteLength;
+        }
+    }
+
+    public write(buffer: ArrayBuffer): void {
+        Contracts.throwIfNullOrUndefined(this.privAudioFormat, "must set format before writing.");
+        if (this.privFd !== undefined) {
+            fs.writeSync(this.privFd, new Int8Array(buffer), 0, buffer.byteLength, this.privOffset);
+            this.privOffset += buffer.byteLength;
```

Comment:
this should be the length used, but writeSync also returns the bytes read - wondering if that's a safer convention to use?

Reply:
Thanks for pointing this, it's not safe indeed. I will try to use `fs.WriteStream` instead.